python/xir/program.py

```diff
@@ -2,31 +2,40 @@
 import warnings
 from decimal import Decimal
 from typing import Any, Dict, List, Sequence, Set, Tuple, Union
+from numbers import Number
 
 from .decimal_complex import DecimalComplex
 from .utils import strip
 
 """This module contains the XIRProgram class and classes for the Xanadu IR"""
 
+Wire = Union[int, str]
+Param = Union[complex, str, bool, List["Param"]]
```

Comment:
Does this cover `Decimal` and `ComplexDecimal` objects?  Also, why not use `Number` instead of `complex`?

Reply:
Using `Number` doesn't seem to work with PyLance (and not with [Mypy either](https://github.com/python/mypy/issues/3186)), but you're right. `Decimal` and `DecimalComplex` need to be added here as well. They are neither `complex` nor each other in any way.

Reply:
> Using `Number` doesn't seem to work with PyLance (and not with [Mypy either](https://github.com/python/mypy/issues/3186)), but you're right.

😢

> `Decimal` and `DecimalComplex` need to be added here as well. They are neither `complex` nor each other in any way.

Okay, sounds good!